docs/source/en/modular_diffusers/getting_started.md

```diff
@@ -0,0 +1,1393 @@
+  null,  # class (same as model_index.json)
+  {      # loading specs map (unique to modular_model_index.json)
+    "repo": "stabilityai/stable-diffusion-xl-base-1.0",  # can be a different repo
+    "revision": null,
+    "subfolder": "text_encoder",
+    "type_hint": [  # (library, class) for the expected component class
+      "transformers",  
+      "CLIPTextModel"
+    ],
+    "variant": null
+  }
+],
+```
+Some components may not have `repo` field, they cannot be loaded from a repository and can only be created with default config from the pipeline
+
+```py
+  "image_processor": [
+    "diffusers",
+    "VaeImageProcessor",
+    {
+      "type_hint": [
+        "diffusers",
+        "VaeImageProcessor"
+      ]
+    }
+  ],
+```
+Unlike standard repositories where components must be in subfolders within the same repo, modular repositories can fetch components from different repositories based on the `loading_specs` dictionary. e.g. the `text_encoder` component will be fetched from the "text_encoder" folder in `stabilityai/stable-diffusion-xl-base-1.0` while other components come from different repositories.
+
+
+### Creating a `ModularPipeline` from `ModularPipelineBlocks`
+
+Each `ModularPipelineBlocks` has an `init_pipeline` method that can initialize a `ModularPipeline` object based on its component and configuration specifications.
+
+Let's convert our `t2i_blocks` (which we created earlier) into a runnable `ModularPipeline`:
+
+```py
+# We already have this from earlier
+t2i_blocks = SequentialPipelineBlocks.from_blocks_dict(TEXT2IMAGE_BLOCKS)
+
+# Now convert it to a ModularPipeline
+modular_repo_id = "YiYiXu/modular-loader-t2i"
+t2i_pipeline = t2i_blocks.init_pipeline(modular_repo_id)
+```
+
+<Tip>
+
+💡 We recommend using `ModularPipeline` with Component Manager by passing a `components_manager`:
+
+```py
+>>> components = ComponentsManager()
+>>> pipeline = blocks.init_pipeline(modular_repo_id, components_manager=components)
+```
+
+This helps you to:
+1. Detect and manage duplicated models (warns when trying to register an existing model)
+2. Easily reuse components across different pipelines
+3. Apply offloading strategies across multiple pipelines
+
+You can read more about Components Manager [here](TODO)
+
+</Tip>
+
+
+### Creating a `ModularPipeline` with `from_pretrained`
+
+You can create a `ModularPipeline` from a HuggingFace Hub repository with `from_pretrained` method, as long as it's a modular repo:
+
+```py
+# YiYi TODO: this is not yet supported actually 😢, need to add support
+from diffusers import ModularPipeline
+pipeline = ModularPipeline.from_pretrained(repo_id, components_manager=..., collection=...)
+```
+
+Loading custom code is also supported:
+
+```py
+from diffusers import ModularPipeline
+modular_repo_id = "YiYiXu/modular-diffdiff"
+diffdiff_pipeline = ModularPipeline.from_pretrained(modular_repo_id, trust_remote_code=True)
+```
+
+### Loading components into a `ModularPipeline`
```

Comment:
So far, we have shown how to perform loads from a repo id but the reader might also be curious about how those repositories were written.

Should we cover saving as well?

Reply:
saving was talked about in write your custom pipeine part - it is more relevant there